src/commands/appSettings/filterDownloadAppSettings.ts

```diff
@@ -11,42 +11,75 @@ import { localize } from "../../localize";
+    let overwriteSettingsToIgnore: boolean = false;
 
     const addedKeys: string[] = [];
     const updatedKeys: string[] = [];
     const userIgnoredKeys: string[] = [];
     const matchingKeys: string[] = [];
+    const securitySettingsIgnored: string[] = [];
+    const listOfSettingsToIgnore: string[] = ["AzureWebJobsStorage", "WEBSITE_CONTENTAZUREFILECONNECTIONSTRING", "WEBSITE_CONTENTSHARE"];
 
     for (const key of Object.keys(sourceSettings)) {
-        if (destinationSettings[key] === undefined) {
-            addedKeys.push(key);
-            destinationSettings[key] = sourceSettings[key];
-        } else if (destinationSettings[key] === sourceSettings[key]) {
-            matchingKeys.push(key);
-        } else if (sourceSettings[key]) { // ignore empty settings
-            if (!suppressPrompt) {
+        if (listOfSettingsToIgnore.includes(key)) {
+            if (!suppressPromptOfSettingsToIgnore) {
                 const yesToAll: vscode.MessageItem = { title: localize('yesToAll', 'Yes to all') };
                 const noToAll: vscode.MessageItem = { title: localize('noToAll', 'No to all') };
-                const message: string = localize('overwriteSetting', 'Setting "{0}" already exists in "{1}". Overwrite?', key, destinationName);
+                const message: string = localize('overwriteSetting', 'Setting "{0}" has been identified as unsafe. Download?', key);
                 const result: vscode.MessageItem = await context.ui.showWarningMessage(message, { modal: true }, DialogResponses.yes, yesToAll, DialogResponses.no, noToAll);
                 if (result === DialogResponses.yes) {
-                    overwriteSetting = true;
+                    overwriteSettingsToIgnore = true;
                 } else if (result === yesToAll) {
-                    overwriteSetting = true;
-                    suppressPrompt = true;
+                    overwriteSettingsToIgnore = true;
+                    suppressPromptOfSettingsToIgnore = true;
                 } else if (result === DialogResponses.no) {
-                    overwriteSetting = false;
+                    overwriteSettingsToIgnore = false;
                 } else if (result === noToAll) {
-                    overwriteSetting = false;
-                    suppressPrompt = true;
+                    overwriteSettingsToIgnore = false;
+                    suppressPromptOfSettingsToIgnore = true;
                 }
-            }
 
-            if (overwriteSetting) {
-                updatedKeys.push(key);
+                if (overwriteSettingsToIgnore) {
+                    updatedKeys.push(key);
+                    destinationSettings[key] = sourceSettings[key];
+                } else {
+                    securitySettingsIgnored.push(key);
+                    destinationSettings[key] = "";
+                }
+            }
+        }
+        else {
+            if (destinationSettings[key] === undefined) {
```

Comment:
`if (!destinationSettings[key])`

Reply:
Agreed.

Reply:
oh I see... in this case, an empty setting will also trigger this condition. In this case, do you mind adding a comment right under `if (destinationSettings[key] === undefined) {` something like `// Have an explicit check for undefined as empty settings should not pass the condition` like this in future someone looking at the code can know why this specific condition was added.